### ISSUE TITLE
Simplify signature of `io.timeout`

### ISSUE THREAD

**Issue by jdegoes** [scalaz/scalaz-zio#325]
1. Rename the current timeout to be `timeout0`.
    
    ```scala
    final def timeout0[Z](z: Z)(f: A => Z)(d: Duration): IO[E, Z]
    ```
2. Introduce a new simplified `timeout`:
    
    ```scala
    final def timeout(d: Duration): IO[E, Option[A]] = timeout0[Option[A]](None)(Some(_))(d)
    ```

3. Introduce an alternate simplified `timeoutFail`:
    
    ```scala
    final def timeoutFail[E1 >: E](e: E1)(d: Duration): IO[E1, A] = IO.flatten(timeout0[IO[E1, A]](IO.fail(e))(IO.now(_))(d))
    ```

**Comment by darbyo** [comment]
Claimed 👍

**Comment by hepin1989** [comment]
change it to private method ？

**Comment by ghostdogpr** [comment]
Fixed by #328

**Comment by jozic** [comment]
i'm sort of late for the party, but why doesn't `timeout` take `Duration.Finite` instead of `Duration`? Is there a good use case where someone wants to use `timeout` with `Infinity` which will turn in no-op in best case?
